src/desktop_notifier/base.py

```diff
@@ -214,114 +209,41 @@ class Notification:
     message: str
     """Notification message"""
 
-    urgency: Urgency
+    urgency: Urgency = Urgency.Normal
     """Notification urgency. Can determine stickiness, notification appearance and
     break through silencing."""
 
-    icon: Icon | None
+    icon: Icon | None = None
     """Icon to use for the notification"""
 
-    buttons: tuple[Button, ...]
+    buttons: tuple[Button, ...] = tuple()
```

Comment:
1. any reason to not just use a `list`?
2. you should probably use a `field(default_factory=list)` (or `default_factory=tuple` if you really prefer) when instantiating a mutable property in a `dataclass`... otherwise you may get unexpected results.

Reply:
haven't tested but just an "a-ha" moment... looking closer suspect (strongly) the lack of a `default_factory` is the reason for [the bug i found](https://github.com/samschott/desktop-notifier/pull/164#issuecomment-2256734616)... reason being that if you don't use a factory method every `Notification` will probably use _the same_ `buttons` tuple.

Reply:
Reasons not to use list: I want notifications to be immutable (once sent, there is little purpose in changing a notification instance and it could lead to unexpected bugs).

Reply:
Python tuples are immutable, and all empty tuples are the same instance. You can never add or remove anything from a tuple, unlike dicts or lists.